libqtile/widget/redshift.py

```diff
@@ -0,0 +1,333 @@
+        try:
+            self._get_rs_driver().enable(self.temperature, gamma_val, self.brightness)
+        except Exception as e:
+            raise RuntimeError(f"redshift: could not enable redshift: {e}")
+
+    @expose_command
+    def click(self):
+        """
+        Either enable or disable the widget ONLY if we are currently on the
+        first line ie. the enabled/disabled text.
+        """
+        if self._line_index != 0:
+            return
+        if self.is_enabled:
+            self.reset_redshift()
+        else:
+            self.enable_redshift()
+        self.is_enabled = not self.is_enabled
+        self._set_text()
+        self.bar.draw()
+
+    @expose_command
+    def reset_redshift(self):
+        """
+        Make a call to redshift to reset everything.
+        """
+        try:
+            self._get_rs_driver().reset()
+        except Exception as e:
+            raise RuntimeError(f"redshift: could not reset redshift: {e}")
+
+    def _scroll(self, step):
+        """
+        Scroll up/down dictated by 'step' the items
+        and then display that item.
+        """
+        # Make sure that the lines var is set, maybe
+        # some of the values have been updated?
+        self._set_lines()
+        self._line_index = (self._line_index + step) % len(self._lines)
+        self.show_line()
+
+    def _set_lines(self):
+        """
+        Set the list of formatted text items to scroll through.
+        """
+        first_text = ""
+        if self.is_enabled:
+            first_text = self.enabled_txt
+        else:
+            first_text = self.disabled_txt
+
+        first_text = self._format_first_text(str(first_text))
+
+        self._lines = [
+            first_text,
+            self._format_brightness_text(),
+            self._format_temp_text(),
+            self._format_gamma_text(
+                GammaGroup(self.gamma_red, self.gamma_green, self.gamma_blue)
+            ),
+        ]
+
+    def _set_text(self):
+        """
+        Set text (not update) to the current index of lines.
+        """
+        if self._line_index == 0:
+            if self.is_enabled:
+                self.text = self._format_first_text(str(self.enabled_txt))
+            else:
+                self.text = self._format_first_text(str(self.disabled_txt))
+        else:
+            self.text = self.format_object(self._lines[self._line_index])
+
+    def _format_temp_text(self) -> str:
+        """
+        Format the temperature text.
+        """
+        return str(self.temperature_fmt).format(temperature=self.temperature)
+
+    def _format_brightness_text(self) -> str:
+        """
+        Format the brightness text.
+        """
+        return str(self.brightness_fmt).format(brightness=self.brightness)
+
+    def _format_gamma_text(self, gamma: GammaGroup) -> str:
+        """
+        Format the gamma text.
+        """
+        return gamma._format_gamma(str(self.gamma_fmt))
```

Comment:
Why wouldn't `self.gamma_fmt` be a string?

Reply:
Format comment above